### PR TITLE
Add sign in with Apple

### PR DESCRIPTION
**Pull request by belcherj**
#### Changes proposed in this Pull Request

* Adds Sign in with Apple

#### Testing instructions
(This is behind a flag, just need to test Google Sign in isn't broken)
* Test that you can log in with Google

Screen Shot 2019-08-05 at 2 40 07 PM

### CODE REVIEW

File: client/components/social-buttons/apple.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+					)
+			)
+			.catch( error => {
+				this.initialized = null;
+
+				if ( 'idpiframe_initialization_failed' === error.error ) {
+					// This error is caused by 3rd party cookies being blocked.
+					this.setState( {
+						error: translate(
+							'Please enable "third-party cookies" to connect your Apple account.'
```
Comment (a8ci18n) [inline client/components/social-buttons/apple.js]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Please enable "third-party cookies" to connect your Google account.' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/components/social-buttons/style.scss [hunk 2]
```diff
@@ -10,3 +10,18 @@
 		max-width: 200px;
 	}
 }
+
+.apple__signin-button div {
```
Comment (griffbrad) [inline client/components/social-buttons/style.scss]: Could you adjust this class name to match Calypso naming conventions? (https://wpcalypso.wordpress.com/devdocs/docs/coding-guidelines/css.md)

File: client/blocks/login/social.jsx [hunk 3]
```diff
@@ -97,44 +98,51 @@ class SocialLoginForm extends Component {
...
+					login( { isNative: true, socialService: service } ) }`
 			: null;
+	};
 
+	render() {
+		const { redirectTo, uxMode } = this.props;
 		return (
 			<Card className="login__social">
 				<div className="login__social-buttons">
 					<GoogleLoginButton
```
Comment (griffbrad) [inline client/blocks/login/social.jsx]: Should the wrapping `div` from `GoogleLoginButton` be moved here so it wraps both Google and Apple?

File: client/components/social-buttons/style.scss (hunk 2 above)
Comment (belcherj) [reply client/components/social-buttons/style.scss]: Linter said `apple__` or `social-buttons__` was fine. Switched to `social-buttons__apple-button` as I think that is most in line with the convention.

File: client/blocks/login/social.jsx (hunk 3 above)
Comment (belcherj) [reply client/blocks/login/social.jsx]: I deleted that enclosing div as it was unneeded

**Pull request merged**